src/storage/db.rs

```diff
@@ -142,14 +148,17 @@ impl LockerInterface for Storage {
             .get_result(&mut conn)
             .await;
 
-        output
-            .map_err(|error| match error {
-                diesel::result::Error::NotFound => error::StorageError::NotFoundError,
-                _ => error::StorageError::FindError,
-            })
-            .map_err(error::ContainerError::from)
-            .map_err(From::from)
-            .map(|inner| inner.into())
+        let output = match output {
+            Err(err) => match err {
+                diesel::result::Error::NotFound => {
+                    Err(err).change_error(error::StorageError::NotFoundError)
+                }
+                _ => Err(err).change_error(error::StorageError::FindError),
+            },
+            Ok(locker) => Ok(locker),
+        };
+
+        output.map_err(From::from).map(|inner| inner.into())
```

Comment:
```suggestion
        output.map_or_else(From::from, From::from)
```

Reply:
Here the output requires type conversion for both Ok and Err variant. map_or_else would have worked for Option